pkg/apis/cluster/v1alpha1/common_types.go

```diff
@@ -39,5 +42,25 @@ type ProviderConfig struct {
 // ProviderConfigSource represents a source for the provider-specific
 // resource configuration.
 type ProviderConfigSource struct {
-	// TODO(roberthbailey): Fill these in later
+	// The machine class from which the provider config should be sourced.
+	// +optional
+	MachineClass *MachineClassRef `json:machineClass,omitempty`
+}
+
+// MachineClassRef is a reference to the MachineClass object. Controllers should find the right MachineClass using this reference.
+type MachineClassRef struct {
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
```

Comment:
Isn't there a "reference" type that we should use instead of objectmeta? Something like LocalReference or ObjectReference?

Reply:
Yes, my bad, made it objectreference.